### ISSUE TITLE
Add ability to get entire child entities

### ISSUE THREAD

**Issue by infosia** [OurMachinery/themachinery-public#400]
`tm_entity_api` has `entities_matching()` but it requires component mask. Is there any way to retrieve entire child entities? I expected something like

```
tm_entity_set_t *(*entities)(tm_entity_context_o *ctx);
```

Or 

```
// carray
tm_entity_t *(*children)(tm_entity_context_o *ctx);
```

**Comment by niklas-ourmachinery** [comment]
Yes! Children actually get added to an `tm_owner_component` in the entity and you can use the `tm_owner_component_api` to query for them:

~~~c
tm_owner_component_api->children(...);
~~~

**Comment by infosia** [comment]
@niklas-ourmachinery Thanks for quick reply! honestly I was confused tm_entity_api  has no `children()` while there's `parent()`, and there's `num_entities()` while nothing equivalent of `entities(i)`.

**Comment by niklas-ourmachinery** [comment]
I agree, it is confusing... I think since the "Owner Component" is kind of "hidden" -- it makes sense to expose this functionality in `tm_entity_api` too, to not confuse people. Reopening this ticket to track that task.

**Comment by niklas-ourmachinery** [comment]
Fixed for 2020.12
